packages/mcp/src/tools/process-manager.ts

```diff
@@ -0,0 +1,596 @@
+  name?: string;
+  status: TaskStatus;
+  cwd?: string;
+  env?: Record<string, string>;
+  pid: number | null;
+  startedAt?: Date;
+  completedAt?: Date;
+  exitCode?: number | null;
+  signal?: NodeJS.Signals | null;
+  timeoutMs?: number;
+  timeoutHandle?: NodeJS.Timeout;
+  process?: ReturnType<typeof spawn>;
+  stdoutText: string;
+  stderrText: string;
+  stdoutLines: string[];
+  stderrLines: string[];
+  stdoutRemainder: string;
+  stderrRemainder: string;
+};
+
+type TaskSummary = Readonly<{
+  id: string;
+  name?: string;
+  command: string;
+  args: readonly string[];
+  status: TaskStatus;
+  pid: number | null;
+  createdAt: string;
+  startedAt?: string;
+  completedAt?: string;
+  exitCode?: number | null;
+  signal?: NodeJS.Signals | null;
+}>;
+
+type RunnerConfig = Readonly<{
+  path: string;
+  maxRunning: number;
+  timeout?: number;
+}>;
+
+let taskCounter = 0;
+let config: RunnerConfig = {
+  path: process.cwd(),
+  maxRunning: 1,
+};
+const tasks = new Map<string, Task>();
+const waitingQueue: string[] = [];
+const runningTasks = new Set<string>();
+const completedTasks: string[] = [];
+
+const buildSummary = (task: Task): TaskSummary => ({
+  id: task.id,
+  name: task.name,
+  command: task.command,
+  args: task.args,
+  status: task.status,
+  pid: task.pid,
+  createdAt: task.createdAt.toISOString(),
+  startedAt: task.startedAt?.toISOString(),
+  completedAt: task.completedAt?.toISOString(),
+  exitCode: task.exitCode,
+  signal: task.signal ?? undefined,
+});
+
+const getConfig = () => ({ ...config });
+
+const updateConfig = (key: string, value: string | number | undefined) => {
+  if (key === "path") {
+    if (typeof value !== "string" || !value) {
+      throw new Error("path must be a non-empty string");
+    }
+    config = { ...config, path: path.resolve(value) };
+    return config;
+  }
+  if (key === "maxRunning") {
+    const parsed = Number(value);
+    if (!Number.isInteger(parsed) || parsed <= 0) {
+      throw new Error("maxRunning must be a positive integer");
+    }
+    config = { ...config, maxRunning: parsed };
+    maybeRunNext();
+    return config;
+  }
+  if (key === "timeout") {
+    if (value === undefined || value === null) {
+      const { timeout, ...rest } = config;
+      config = rest as RunnerConfig;
+      return config;
+    }
+    const parsed = Number(value);
+    if (!Number.isFinite(parsed) || parsed <= 0) {
+      throw new Error("timeout must be a positive number of milliseconds");
+    }
+    config = { ...config, timeout: parsed };
+    return config;
+  }
+  throw new Error(`Unknown config key: ${key}`);
+};
+
+const appendOutput = (task: Task, kind: "stdout" | "stderr", chunk: Buffer) => {
+  const text = chunk.toString("utf8");
+  if (kind === "stdout") {
+    task.stdoutText += text;
+    const combined = task.stdoutRemainder + text;
+    const parts = combined.split(/\r?\n/);
+    task.stdoutRemainder = parts.pop() ?? "";
+    task.stdoutLines.push(...parts);
```

Comment:
Prefer finalization on `"close"` (after stdio is flushed) rather than `"exit"`. We can still keep an `error` handler and a fallback on `exit` if needed, but `close` prevents tail loss under heavy stdout/stderr.